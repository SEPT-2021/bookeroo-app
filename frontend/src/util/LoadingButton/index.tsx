import React from "react";
import { Button, ButtonProps, CircularProgress } from "@material-ui/core";

function LoadingButton({ loading, ...rest }: LoadingButtonProps) {
  return (
    <Button disabled={loading} {...rest}>
      {loading ? (
        <CircularProgress size={24} color="secondary" />
      ) : (
        rest.children
      )}
    </Button>
  );
}

interface LoadingButtonProps extends ButtonProps {
  loading?: boolean;
}

export default LoadingButton;
