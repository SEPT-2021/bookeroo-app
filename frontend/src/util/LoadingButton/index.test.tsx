import React from "react";
import { render } from "@testing-library/react";
import LoadingButton from ".";

test("renders loading", () => {
  const { container } = render(<LoadingButton loading>Example</LoadingButton>);
  expect(container).toMatchSnapshot();
});

test("renders default", () => {
  const { container } = render(<LoadingButton>Example</LoadingButton>);
  expect(container).toMatchSnapshot();
});
