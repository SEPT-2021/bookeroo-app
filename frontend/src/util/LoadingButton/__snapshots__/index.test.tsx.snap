// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<div>
  <mock-loadingbutton>
    Example
  </mock-loadingbutton>
</div>
`;

exports[`renders loading 1`] = `
<div>
  <mock-loadingbutton
    loading="true"
  >
    Example
  </mock-loadingbutton>
</div>
`;
